### PR TITLE
AP_Arming: MAVLINK arm ignores radio stick operation

### LINKED ISSUE #20995: Copter: unable to arm when throttle is maximized in GUIDED mode
## Bug report

**Issue details**

I did an ARM from a QGC in GUIDED mode with the throttle of the radio on top.
I could not arm with "PreArm: Throttle (RC3) is not neutral".
I expect it to ignore the radio input except for the flight mode channel.
I can transition from GUIDED mode to other modes by changing the flight mode channel.

**Version**

ArduCopter V4.3.0-dev (e9942f9a)

**Platform**
[  ] All
[  ] AntennaTracker
[X] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**

QUAD COPTER

**Hardware type**

SITL]

**Logs**

Screenshot from 2022-06-18 01-29-16

### PR DESCRIPTION
**Pull request by muramura**
I resolve #20995 .

I checked with SITL.
I maxed out the throttle and issued ARM instructions from the QGC.
I was able to confirm that it would arm.

### CODE REVIEW

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -651,6 +651,12 @@ bool AP_Arming::rc_arm_checks(AP_Arming::Method method)
         check_failed(ARMING_CHECK_PARAMETERS, true, "Mode channel and RC%d_OPTION conflict", rc().flight_mode_channel_number());
         check_passed = false;
     }
+
+    // The MAVLINK command arm does not check radio input.
```
Comment (peterbarker) [inline libraries/AP_Arming/AP_Arming.cpp:655]: This doesn't look right; where's the check for which mode you're in?

Comment (muramura) [reply libraries/AP_Arming/AP_Arming.cpp:655]: @peterbarker san.  Thanks for the comment.
When I access the AP_ARMING check from the copter ARMING check, I check the current flight mode.
The AP_ARMING check does not use the argument METHOD.
I added GUIDED to this METHOD.
I changed it to DO NOT CHECK RADIO INPUT if GUIDED.

AFTER
Screenshot from 2022-06-18 10-47-48

**Review by khancyr — COMMENTED** [review]
Why do we need to check radio in guided ? That isn't right

**Comment by WickedShell** [comment]
I'm not a major copter person, but I'd argue that the stick check is still valid. The stick check is designed to catch the case when you have nudged trims off, or severely messed up your transmitter. I don't see why we wouldn't want this same check in guided. It seems to be removing a level of protection as far as I can see.

**Comment by muramura** [comment]
@WickedShell san. Comments thanks.

I want to use cc mainly and fc as a subordinate in guided mode.  This merit is that all judgments can be made with cc.  You can check it because you can get it by telemetry in the state of the stick with cc.  If you want to detect an abnormality in the radio, you should switch to the guided mode with the flight mode sw when the operator recognizes that you have armed and armed in a mode other than the guided mode.

**Comment by WickedShell** [comment]
If you are operating without any RC system then why not just disable the check via the RC_OPTIONS parameter? If you are going to use RC at any point then the check is helping you, and should be left in.

**Comment by khancyr** [comment]
I agree with WickedShell statement. The check should be keept in all case. It will prevent some stupid mistake when control is done by CC by still get an operator in the loop.

If the radio isn't plugged, the check is skipped anyway.
So the only case where it matter is when a radio is there. In that case it will prevent accident doing ARM in GUIDED and switch to a radio control mode. If the stick was full throttle, the copter will takeoff directly full throttle which is really dangerous.

**Comment by khancyr** [comment]
Unless, you can demonstrate a blocking point or issue with the check, I will close this next week.
@rmackay9 any objection ? I don't want to bring the language barrier on the PR, but I don't want to miss something as this is safety related.

**Comment by muramura** [comment]
I think of GUIDED mode as a flight mode where the CC has complete control over the FC.
Complete means that the FC does not make decisions in response to external stimuli, but the CC does.
I think that the phenomenon of not being able to arm while the radio is in operation is not good for the CC.
If you want to check the status of the radio, you can arm in a mode other than GUIDED mode and switch to GUIDED mode after arming.

**Comment by peterbarker** [comment]
> The radio is there for the FC for anomalies.

Yes.  And it would be a very bad thing if the throttle was at maximum when that anomaly occured!

OTOH - it is almost always a bad thing if your throttle is at zero, too, so actually enforcing a zero-throttle in an arming check is actually kind of bad.

@muramura what if you enabled the sprung-throttle mode?  I believe that lets you arm when the throttle is mid-stick?  This would mean that if you *did* have an anomaly then switching the mode with the transmitter would not instantly kill the aircraft (or send it into the ceiling)

**Comment by WickedShell** [comment]
> it is almost always a bad thing if your throttle is at zero, too, so actually enforcing a zero-throttle in an arming check is actually kind of bad.

I think it's late and I'm missing something obvious, but what's the bad thing that happens if throttle is at 0?

**Comment by rmackay9** [comment]
I agree with the comments that this is probably not the correct implementation but in past releases at least we have not checked the pilot throttle when arming in Guided from the GCS.

The code that should lead to this skipped checked is linked from issue https://github.com/ArduPilot/ardupilot/issues/21415 and [here's a direct link](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/AP_Arming.cpp#L633).

These checks are skipped because some pilots want to leave the RC input throttle high while flying in Guided mode so that if they need to switch to a more manual mode the vehicle stays in a hover (or even climbs a bit) rather than falling rapidly.

I think that @muramura has discovered that this check is no longer skipped as it should be.

Apologies if I'm misunderstanding something

**Comment by peterbarker** [comment]
We have an RC options bit that does this now

**Pull request closed without merging**